addons/sale_stock/models/stock.py

```diff
@@ -155,14 +157,13 @@ class ProductionLot(models.Model):
 
     @api.depends('name')
     def _compute_sale_order_ids(self):
+        sale_orders = defaultdict(lambda: self.env['sale.order'])
+        for move_line in self.env['stock.move.line'].search([('lot_id', 'in', self.ids), ('state', '=', 'done')]):
+            move = move_line.move_id
+            if move.state == 'done' and move.picking_id.location_dest_id.usage == 'customer' and move.sale_line_id.order_id:
```

Comment:
if a move line is in state 'done', the stock move is also in that state (they are related field) 
for the rest, your patch is good. 
Thanks